### PR TITLE
Add CheckpointConnector internal commentaries

### PR DESCRIPTION
**Pull request by tarepan**
## What does this PR do? -> 📃 
Add internal commentaries toward CheckpointConnector and PyTorch-Lightning save/load.  
This PR will help future committers who hope to engineer CheckpointConnector.  

p.s. No code modified.

## Before submitting
- [n/a] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [n/a] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [n/a] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.    
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 👍

### CODE REVIEW

File: pytorch_lightning/trainer/connectors/checkpoint_connector.py [hunk 1]
```diff
@@ -44,6 +44,18 @@
     OMEGACONF_AVAILABLE = True
 
 
+# [Save & Load scheme]
```
Comment (tchaton) [inline pytorch_lightning/trainer/connectors/checkpoint_connector.py:47]: It is a great idea, but I have some trouble to understand what you meant to explain. Would mind refactoring this comment ?

File: pytorch_lightning/trainer/connectors/checkpoint_connector.py [hunk 2]
```diff
@@ -241,15 +256,31 @@ def hpc_save(self, folderpath: str, logger):
...
     def dump_checkpoint(self, weights_only: bool = False) -> dict:
-        """Creating model checkpoint.
+        """Creating a model checkpoint dictionary object from various component states.
 
         Args:
             weights_only: saving model weights only
 
         Return:
-             structured dictionary
+            structured dictionary: {
```
Comment (tchaton) [inline pytorch_lightning/trainer/connectors/checkpoint_connector.py:265]: Great addition !

**Review by tchaton — CHANGES_REQUESTED** [review]
Great PR ! Some clarification with first comment :)

File: pytorch_lightning/trainer/connectors/checkpoint_connector.py (hunk 1 above)
Comment (tarepan) [reply pytorch_lightning/trainer/connectors/checkpoint_connector.py:47]: What I hope to explain is "how states is saved, then loaded".  

- save: various states -> checkpoint dict -> .ckpt file
- load: .ckpt file -> checkpoint dict -> various states 

And also hope to explain "which function implement each steps".  

- save: `save_checkpoint` (`dump_checkpoint` for making dict & `atomic_save` for write) 
- load: `restore` (`pl_load` for read & part of `restore` for restoring states)

But the scheme graph looks complex as you mentioned.  
**I will simply write above comment in each function.**

Comment (tarepan) [reply pytorch_lightning/trainer/connectors/checkpoint_connector.py]: Refactored.

**Review by Borda — APPROVED** [review]
lgtm

**Review by tchaton — APPROVED** [review]
Great addition !

**Review by SeanNaren — APPROVED** [review]

**Pull request merged**